numpy/core/tests/test_multiarray.py

```diff
@@ -1760,32 +1760,34 @@ def test_sort(self):
                 msg = 'byte-swapped complex sort, dtype={0}'.format(dt)
                 assert_equal(c, arr, msg)
 
-        # test string sorts.
-        s = 'aaaaaaaa'
-        a = np.array([s + chr(i) for i in range(101)])
+    @pytest.mark.parametrize('message, s, dtype',  [('string', 'aaaaaaaa', 'str'),('unicode', 'aaaaaaaa', np.unicode_)])
+    def test_sort_string(self, message, s, dtype):
+        a = np.array([s + chr(i) for i in range(101)], dtype = dtype)
         b = a[::-1].copy()
         for kind in self.sort_kinds:
-            msg = "string sort, kind=%s" % kind
+            msg = message + " sort, kind=%s" % kind
             c = a.copy()
             c.sort(kind=kind)
             assert_equal(c, a, msg)
             c = b.copy()
             c.sort(kind=kind)
             assert_equal(c, a, msg)
 
-        # test unicode sorts.
-        s = 'aaaaaaaa'
-        a = np.array([s + chr(i) for i in range(101)], dtype=np.unicode_)
-        b = a[::-1].copy()
-        for kind in self.sort_kinds:
-            msg = "unicode sort, kind=%s" % kind
+    @pytest.mark.parametrize('message, dtype', [('datetime64', 'datetime64[D]'), ('timedelta64', 'timedelta64[D]')])
+    def test_sort_time(self, message, dtype):
+        # test datetime64 and timedelta64 sorts.
+        a = np.arange(0, 101, dtype=dtype)
+        b = a[::-1]
+        for kind in ['q', 'h', 'm']:
+            msg = message + "sort, kind=%s" % kind
             c = a.copy()
             c.sort(kind=kind)
             assert_equal(c, a, msg)
             c = b.copy()
             c.sort(kind=kind)
             assert_equal(c, a, msg)
 
+    def test_sort_object(self):
```

Comment:
Half way down this function would be a good place for `def test_sort_structured(self):`